pkgs/applications/networking/mailreaders/hey-mail/default.nix

```diff
@@ -0,0 +1,114 @@
+{ fetchurl, lib, stdenv, squashfsTools, makeWrapper, c-ares, gtk3-x11,
+glib, libevent, libdrm, libvpx, libxslt, libnotify,
+libappindicator-gtk2, libappindicator-gtk3, atk, mesa, cups, systemd,
+alsaLib, at-spi2-atk, at-spi2-core, gdk-pixbuf, gnome2, cairo, xorg,
+ffmpeg, http-parser, nss, nspr, dbus, expat }:
+
+let
+  deps = [
+    c-ares
+    gtk3-x11
+    glib
+    libevent
+    libdrm
+    libvpx
+    libxslt
+    libnotify
+    libappindicator-gtk2
+    libappindicator-gtk3
+    atk
+    mesa
+    cups
+    systemd
+    alsaLib
+    at-spi2-atk
+    at-spi2-core
+    gdk-pixbuf
+    gnome2.pango
+    cairo
+    xorg.libxcb
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrender
+    xorg.libXtst
+    xorg.libXrandr
+    ffmpeg
+    http-parser
+    nss
+    nspr
+    dbus
+    expat
+    stdenv.cc.cc
+  ];
+  rev = "10";
+
+in
+
+stdenv.mkDerivation rec {
+  pname = "hey-mail";
+  version = "1.1.0";
+
+  src = fetchurl {
```

Comment:
It would be nice to have a comment here explaining where to scratch the download link, or better, add to the package an `updateScript` that would do it automatically, see an example for `zoom-us`.

Reply:
@doronbehar  The idea here is that the `updateScript` will pull info from the download link like `version` and `ref` dynamically? So if we trigger update it will try to build the package with the latest version available in the download link?

Reply:
`zoom-us` has:

https://github.com/NixOS/nixpkgs/blob/560cc2af818e6f08c5ef316d2f1dedfa47aee51a/pkgs/applications/networking/instant-messengers/zoom-us/default.nix#L119

And this enables anyone to update it with:

```
nix-shell maintainers/scripts/update.nix --argstr package zoom-us --argstr commit true
```

Which with a certain mediator calls the update script which has:

https://github.com/NixOS/nixpkgs/blob/560cc2af818e6f08c5ef316d2f1dedfa47aee51a/pkgs/applications/networking/instant-messengers/zoom-us/update.sh#L1-L10

-----

Overall, this is a sophisticated way of showing anyone interested not only where the version of the package appears in upstream's website, it also automates the process of retrieving the version string and updating the nix expression of the package. It doesn't build the package you updated but with `--argstr commit true` it runs a `git commit` with a proper `--message` suitable for the change in version done, if any. In your case it might be a bit more complicated because as I see it, you have to update 3-4 strings in the expression:

1. The version.
2. The `rev` maybe?
3. The hash that's part of the URL.
4. The `sha256` of the downloaded file.

`update-source-version` usually updates easily only `version` and `sha256` of the downloaded link. In your case it might get complicated. If you think it's above your Nix / shell skills level, at least please make the effort of explaining in a comment what is `rev` and where do you find the download link (I didn't find it easily) and what is the source of the small hash in the URL, if it's likely to change in the next version of the package.